### PR TITLE
AB Email Notifications - Block validation fix

### PR DESCRIPTION
**Pull request by HasnatL-Adapptlimited**
# Links
[Base PR](https://github.com/ReliefApplications/ems-frontend/pull/2460)

In reference to point 3:
https://github.com/ReliefApplications/ems-frontend/pull/2460#issuecomment-2072440406

# Tickets
- [ ] If user selects a resource but no fields they cant see a preview of the block or proceed to next step until its fixed

### CODE REVIEW

**Comment by HasnatL-Adapptlimited** [comment]
Hey @AntoineRelief 
CC: @jayasinp-adappt 

PR has been opened for point 3 in the high priority comments

File: apps/front-office/src/app/application/application.component.ts [hunk 1]
```diff
@@ -218,19 +218,19 @@ export class ApplicationComponent
...
-    // }
+    if (
+      this.ability.can(
+        'manage',
+        subject('Template', { application: application.id })
+      )
+    ) {
+      // if can manage apps / can manage email notifications in app
+      this.adminNavItems.push({
+        name: this.translate.instant('common.email.notification.few'),
```
Comment (AntoineRelief) [inline apps/front-office/src/app/application/application.component.ts:229]: @HasnatL-Adapptlimited 
this was already fixed in the main branch, so I'll remove all changes that relates to that

File: apps/front-office/src/app/application/application.component.ts [hunk 2]
```diff
@@ -218,19 +218,19 @@ export class ApplicationComponent
...
-    //   });
-    // }
+    if (
+      this.ability.can(
+        'manage',
+        subject('Template', { application: application.id })
+      )
+    ) {
+      // if can manage apps / can manage email notifications in app
+      this.adminNavItems.push({
```
Comment (AntoineRelief) [inline apps/front-office/src/app/application/application.component.ts]: @HasnatL-Adapptlimited 

I thought we agreed not to deploy it in the front-office in phase 1
Am I wrong?

Comment (HasnatL-Adapptlimited) [reply apps/front-office/src/app/application/application.component.ts]: Hey @AntoineRelief 
This is correct, you are not wrong I will rebase the branch and correct this

**Comment by AntoineRelief** [comment]
@jayasinp-adappt @HasnatL-Adapptlimited 

I can't merge it as it is, as it creates a weird issue

See video below
The block should be valid, but I'm now stuck on the block edition page due to that
I tried on the main branch, and couldn't reproduce it. It seems the issue is created by this new PR

https://www.loom.com/share/e12e4e646a3a4d2b8dcc4f263978c5f4?sid=43654531-88c3-47cf-9ac4-e281c6540436

**Comment by HasnatL-Adapptlimited** [comment]
Hi @AntoineRelief 
CC: @jayasinp-adappt 

- Validation Fixed
- Removed email notifications unhide commit

**Comment by AntoineRelief** [comment]
@HasnatL-Adapptlimited @jayasinp-adappt 

Merged this one 👍 

I only changed the templates & translation strings for dataset alerts, to have cleaner code

**Pull request merged**